hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +264,220 @@ default Table newHmsTable(String hmsTableOwner) {
 
     return newTable;
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  default void commitWithLocking(
```

Comment:
In general, my comments in this method are that, we should keep the existing logic as much as possible, as it is quite delicate and we don't want to re-introduce bugs that have been fixed over the many releases.

Reply:
+1

Reply:
I have copied the same code from HiveTableOperations . Only part has been parameterised is `setHmsParameters` .
CommitFlow is till the same as:
- lock table
- load table if any
- check for concurrent commit
- set HMS parameters without obsolete props
- ensure lock is active
- persist
- ensure lock is active
- commit
- do the error handling with reload table and check commit status
-